src/cmd/file.c

```diff
@@ -523,6 +523,9 @@ Filemagic long0tab[] = {
 	070707,		0xFFFF,		"cpio archive\n", OCTET,
 	0x2F7,		0xFFFF,		"tex dvi\n", "application/dvi\n",
 	0xfffa0000,	0xfffe0000,	"mp3 audio\n",	"audio/mpeg\n",
+	0xcafebabe,	0xFFFFFFFF,	"Mach-O executable\n",	"application/x-mach-binary\n",
+	0xfeedface,	0xFFFFFFFE,	"Mach-O executable\n",	"application/x-mach-binary\n",
+	0xbebafeca,	0xFFFFFFFF,	"Java class\n",	"application/x-java-applet\n",
```

Comment:
% hexdump -C -n 4 HelloWorld.class
00000000  ca fe ba be

http://radio-weblogs.com/0100490/2003/01/28.html

Reply:
Thanks for the link, such history is always interesting to read.
I’m not sure what your point is though?

~~~
% hexdump -C -n4 Main.class
00000000  ca fe ba be                                       |....|
00000004
% 9 file Main.class
Main.class: Java class
% hexdump -C -n4 Main
00000000  cf fa ed fe                                       |....|
00000004
% 9 file Main
Main: Mach-O executable
% uname -m -s
Darwin x86_64
~~~

Reply:
I'd expect 0xcafebabe to be filemagic for a Java class?